assets/js/util.js

```diff
@@ -293,15 +298,74 @@ function callApy(options, endpoint) {
     var requestUrl = window.location.protocol + window.location.hostname +
         window.location.pathname + '?' + $.param(requestOptions.data);
 
+    callApyStartTime = Date.now();
     if(requestUrl.length > THRESHOLD_REQUEST_URL_LENGTH) {
         requestOptions.type = 'POST';
         return $.ajax(requestOptions);
     }
     return $.jsonp(requestOptions);
 }
 
+function callApyDuration() {
+    callApyEndTime = Date.now();
+    checkServiceLoadTimes(callApyEndTime - callApyStartTime);
+}
+
+function checkServiceLoadTimes(requestDuration) {
+    var individualDurationThreshold = 2000;
+    var cumulativeDurationThreshold = 1500;
+    var demoThreshold = 10000; // for demo purposes only
+    if(store.able()) {
+        if(sessionStorage.requestsMade) {
+            sessionStorage.requestsMade = Number(sessionStorage.requestsMade) + 1;
+        }
+        else {
+            sessionStorage.requestsMade = 1;
+        }
+
+        if(sessionStorage.cumulativeRequestsTime) {
+            sessionStorage.cumulativeRequestsTime = Number(sessionStorage.cumulativeRequestsTime) + requestDuration;
+        }
+        else {
+            sessionStorage.cumulativeRequestsTime = requestDuration;
+        }
+
+        var averageRequestsDuration = (sessionStorage.cumulativeRequestsTime) / (sessionStorage.requestsMade);
+
+        if(requestDuration > individualDurationThreshold || averageRequestsDuration > cumulativeDurationThreshold) {
```

Comment:
> Don't worry! The 'clutter' is fine. I want to maximize our time overlap/our time by commenting on more things at one time.

Yes, you will now see a lot of PRs, the ones that were ready in mine since long too :P.

> 1. Retain requests from the last Y minutes.
   2.Retain the last X requests.

And then take the max of these 2?